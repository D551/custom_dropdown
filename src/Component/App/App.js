import React from 'react'
import axios from 'axios';
import './App.scss';
import ManagersDropDown from '../Dropdown/DropDown';
import { formatManagersData } from '../Dropdown/DropDownHelper';
import { Constants } from '../../Constants/lcaleConstants'

function App() {
  const [options, setOptions] = React.useState([]);
  const [selectedValue, setSelectedValue] = React.useState({});

  React.useEffect(() => {
    axios.get(`https://gist.githubusercontent.com/daviferreira/41238222ac31fe36348544ee1d4a9a5e/raw/5dc996407f6c9a6630bfcec56eee22d4bc54b518/employees.json`)
      .then(res => {
        const managers = formatManagersData(res.data);
        setOptions(managers);
      })
  }, []);

  const onSelect = (value) => {
    setSelectedValue(value)
  }

  return (
    <div className="App">
      {options.length &&
        < ManagersDropDown options={options} placeholder={Constants.PLACEHOLDER} onChange={onSelect} />
      }
    </div >
  );
}

export default App;
