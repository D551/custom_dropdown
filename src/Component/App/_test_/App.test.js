import React from 'react'
import { render } from '@testing-library/react';
import App from '../App';

describe('should test App.jsx', () => {
  beforeEach(() => {
    jest.spyOn(React, 'useEffect').mockImplementation((f) => f());
  });
  test('should render App.js', () => {
    const { asFragment } = render(< App />);
    expect(asFragment(< App />)).toMatchSnapshot();
  });
});