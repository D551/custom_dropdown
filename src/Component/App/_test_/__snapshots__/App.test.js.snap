// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App.js 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    0
  </div>
</DocumentFragment>
`;

exports[`should test App.jsx should render App.js 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    0
  </div>
</DocumentFragment>
`;
